Source/missiles.cpp

```diff
@@ -3074,13 +3073,12 @@ void MI_FireNova(Missile &missile)
 		en = TARGET_MONSTERS;
 	}
 	for (const auto &k : VisionCrawlTable) {
-		if (sx1 != k[6] || sy1 != k[7]) {
-			Displacement offsets[] = { { k[6], k[7] }, { -k[6], -k[7] }, { -k[6], +k[7] }, { +k[6], -k[7] } };
-			for (Displacement offset : offsets)
-				AddMissile(src, src + offset, dir, MIS_FIRENOVA, en, id, dam, missile._mispllvl);
-			sx1 = k[6];
-			sy1 = k[7];
-		}
+		if (k[3] == s1)
+			continue;
+		Displacement offsets[] = { k[3], k[3].flipXY(), k[3].flipX(), k[3].flipY() };
+		for (Displacement offset : offsets)
+			AddMissile(src, src + offset, dir, MIS_FIRENOVA, en, id, dam, missile._mispllvl);
+		s1 = k[3];

```

Comment:
These *Nova loops are incredibly wasteful... they're going from `4, 0` to `4, 4` to `0, 4`. Can do that in 9 iterations instead of 23 and drop the `s1` temporary. But don't worry about it in this PR, that's for a future cleanup.

Reply:
opened a pr for this